### DISCUSSION TITLE
v0.21.3+ fullscreen doesn't allow seamless ALT+TAB like in v0.21.2

### DISCUSSION THREAD

**Discussion by ner00** [Stellarium/stellarium#2469] [Q&A]
I'm using Windows 10, and up until now I had not updated from v0.21.1, but after updating to the last version I noticed that every time Stellarium opens in fullscreen or ALT+TAB out of or into it it now transitions with a black screen in between. Previously it was seamless like a borderless fullscreen window mode. I started rolling back the versions until I reached v0.21.2 where the seamless transition into fullscreen returns for the last time, on v0.21.3 it starts behaving as the latest versions.

Note: to achieve this seamless mode in v0.21.2 I have to be running ANGLE mode `--angle-d3d11`.

Is there a specific reason why this new behavior happens? 

Can I make it behave in a manner that resembles borderless window like before post 0.21.2 versions?

Thanks.

**Comment by alex-w** [comment]
Did you try to use any version from series 0.22?

**Reply by ner00** [reply]
Yes, I did, they all behave the same. By the way, I forgot to mention that to achieve this seamless mode in v0.21.2 I have to be running ANGLE mode `--angle-d3d11`.

**Reply by alex-w** [reply]
ANGLE mode is translated OpenGL calls into Direct3D calls and probably you have Intel GPU, because these GPU had issues with OpenGL on Windows. Please try remove current installation the Stellarium and install it again (do "clean installation").

**Reply by ner00** [reply]
I do in fact have an Intel GPU, I don't recall if I chose to use ANGLE due to performance reasons or seamless window transition, or both.

I did as suggested, went as far as using REVO Uninstaller to clean leftover registry keys and files, then clean-installed v0.22.1, unfortunately that didn't make any difference in this regard, both normal mode or ANGLE still have that fullscreen transition every time. I then rolled back to v0.21.2 and the seamless fullscreen works again in ANGLE mode.

Maybe it could be possible to implement a borderless fullscreen/windowed mode that just works out of the box regardless of the GPU?

**Reply by alex-w** [reply]
v0.22.1 and v0.21.2 uses same libraries. Please attach the log from Stellarium.

**Reply by alex-w** [reply]
Please check the drivers for graphics card also!

**Reply by ner00** [reply]
I had updated the graphic drivers before posting on here, but even so the last graphic driver update prior to today was about 2 weeks ago.

Logs from both versions attached below:
[log_v0.21.2.txt](https://github.com/Stellarium/stellarium/files/8791391/log_v0.21.2.txt)
[log_v0.22.1.txt](https://github.com/Stellarium/stellarium/files/8791392/log_v0.22.1.txt)

**Reply by alex-w** [reply]
Version 0.22.1 ignores ANGLE mode, so, this is bug

**Reply by alex-w** [reply]
See bug report: https://github.com/Stellarium/stellarium/issues/2471

**Reply by ner00** [reply]
Thanks for clarifying that, I ran both with `--angle-d3d11 --dump-opengl-details`.
By the way, I tested 80% of the releases between v0.21.1 and 0.22.1, v0.21.3 and onward seem to have the bug if that's the case.

Looking forward to the bugfix of https://github.com/Stellarium/stellarium/issues/2471, thanks for taking the time to follow this up.

**Reply by gzotti** [reply]
Can you try to set an environment variable: QT_OPENGL=angle
before launching Stellarium? Maybe this helps.

**Reply by ner00** [reply]
> Can you try to set an environment variable: QT_OPENGL=angle before launching Stellarium? Maybe this helps.

It did, actually.
Was this variable set temporarily in previous versions at runtime, or was it defined permanently?
I'm still rolling back to v0.21.2, but I can confirm that the variable makes it so that Stellarium acknowledges ANGLE mode.

**Reply by gzotti** [reply]
According to Qt docs, you must either set this variable or set a certain flag during initialisation. We did the latter when the command-line argument is evaluated, but probably some recent change (not yet identified) causes this flag to be coming too late. The environment variable is described in the User Guide, 6.2.2. I have a little notebook from 2010 with pre-Optimus Geforce/Intel (manual switch), on which I have set the angle flag permanently because of outdated (not updated) Geforce drivers. It works, why scrap it?  Another reason ANGLE may be activated automatically is when the GPU is on Qt's blacklist of OpenGL-incapable hardware. We were happy to have solved this issue around 2015, but apparently we must look into it again :-( What worries me is that the Qt6 to which we will have to migrate no longer has ANGLE. I don't want to deprecate all Intel-only pre-2016 PCs like Microsoft did for Win11.

**Reply by gzotti** [reply]
#2471 is fixed.

**Comment by alex-w** [comment]
Please check the fresh version (development snapshot) of Stellarium:
https://github.com/Stellarium/stellarium-data/releases/tag/weekly-snapshot

**Reply by ner00** [reply]
This is weird, it didn't fix the issue discussed here, more on the actual issue https://github.com/Stellarium/stellarium/issues/2471
